File: sort/merge_sort/java/Mergesort.java
public class Mergesort {

    public static int[] intArr = { 45, 11, 32, 9, 21, 20, 6, 5, 17, 13, 12, 9,
            3, 19 };

    public int[] sort(int l, int r) {

        if (l < r) {
            int q = (l + r) / 2;

            sort(l, q);
            sort(q + 1, r);
            merge(l, q, r);
        }
        return intArr;
    }

    private void merge(int l, int q, int r) {
        int[] arr = new int[intArr.length];
        int i, j;
        for (i = l; i <= q; i++) {
            arr[i] = intArr[i];
        }
        for (j = q + 1; j <= r; j++) {
            arr[r + q + 1 - j] = intArr[j];
        }
        i = l;
        j = r;
        for (int k = l; k <= r; k++) {
            if (arr[i] <= arr[j]) {
                intArr[k] = arr[i];
                i++;
            } else {
                intArr[k] = arr[j];
                j--;
            }
        }
    }

    public static void main(String[] args) {
        Mergesort ms = new Mergesort();
        int[] arr = ms.sort(0, intArr.length - 1);
        for (int i = 0; i < arr.length; i++) {
            System.out.println(i + 1 + ": " + arr[i]);
        }
    }
}

File: sort/merge_sort/javascript/MergeSort.js

let merge = function(left, right) {
  let arr = [];

  while (left.length && right.length) {
    if (left[0] < right[0]) {
      arr.push(left.shift());
    } else {
      arr.push(right.shift());
    }
  }
  return arr.concat(left.slice().concat(right.slice()));
};



let mergeSort = function(arr) {
  if (arr.length < 2) {  // don't divide the array 
    return arr;
  }

  const middle = Math.floor(arr.length / 2);
  const left = arr.slice(0, middle); // left part of array
  const right = arr.slice(middle);

  return merge(mergeSort(left), mergeSort(right)); //sort left and right part of array and merge
};

const array = [5, 3, 7, 1, 0, 8, 5];
let result = mergeSort(array.slice()); 
console.log(result)

